File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    c_dict = {}
    for cloth in clothes:
        if cloth[-1] in c_dict:
            c_dict[cloth[-1]] += 1
        else:
            c_dict[cloth[-1]] = 1
            
    for v in c_dict.values():
        answer *= v+1
    
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        Map<String, Integer> closet = new HashMap<>();
        
        // 옷장 수납
        for (String[] pair : clothes) {
            String name = pair[0];
            String category = pair[1];
            
            closet.put(category, closet.getOrDefault(category, 0)+1);
        }
        
        // 경우의 수 계산
        for (String key : closet.keySet()) {
            answer *= (closet.get(key)+1);
        }
        
        return answer-1;
    }
}